nemo/collections/asr/models/label_models.py

```diff
@@ -478,13 +512,37 @@ def verify_speakers(self, path2audio_file1, path2audio_file2, threshold=0.7):
             logging.info(" two audio files are from different speakers")
             return False
 
-    @staticmethod
     @torch.no_grad()
-    def get_batch_embeddings(speaker_model, manifest_filepath, batch_size=32, sample_rate=16000, device='cuda'):
+    def batch_inference(self, manifest_filepath, batch_size=32, sample_rate=16000, device='cuda'):
+        """
+        Perform batch inference on EncDecSpeakerLabelModel. 
+        To perform inference on single audio file, once can use infer_model, get_label or get_embedding
 
-        speaker_model.eval()
-        if device == 'cuda':
-            speaker_model.to(device)
+        To map predicted labels, one can do 
+            `arg_values = logits.argmax(axis=1)`
+            `pred_labels = list(map(lambda t : pred_labels[t], arg_values))`
+
+        Args:
+            manifest_filepath: Path to manifest file
+            batch_size: batch size to perform batch inference
+            sample_rate: sample rate of audio files in manifest file
+            device: compute device to perform operations.
+
+        Returns:
+            All below variables returns values in order (shuffle=False) of audio files in manifest file
+            embs: embeddings of files provided in manifest file
+            logits: logits of final layer of EncDecSpeakerLabel Model
+            gt_labels: labels from manifest file (useful for speaker enrolment and testing)
+            mapped_labels: classification labels in order that mapped from trained model
```

Comment:
Suggest this.
-> Classification labels sorted in the order that they are mapped by the trained model